contract_sale/models/sale_order.py

```diff
@@ -0,0 +1,11 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    contract_id = fields.Many2one(
```

Comment:
I believe that the link between sale orders and contracts should
- via the link between sale order lines and contract lines (already implemented)
- via analytic accounts (not implemented)

Reply:
@sbejaoui We need to be able to select the contract on the sale order to add its sale order lines to the next invoice created by the contract.

Also, products on the sale order lines may not be on the contract lines...

Reply:
so finally, [contract_sale_invoicing](https://github.com/OCA/contract/tree/12.0/contract_sale_invoicing) didn't resolve the use case

Reply:
oops, my bad. All the changes of this PR should be moved to the `contract_sale_invoicing` module.

@murtuzasaleh @nikul-serpentcs